tests/vnc_test.go

```diff
@@ -155,6 +161,54 @@ var _ = Describe("[rfe_id:127][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 			_, err = wrappedRoundTripper.RoundTrip(req)
 			Expect(err).ToNot(HaveOccurred())
 		})
+
+		It("should connect to vnc with --proxy-only flag", func() {
+			By("Invoking virtctl vnc with --proxy-only")
+			proxyOnlyCommand := tests.NewVirtctlCommand("vnc", "--proxy-only", "--namespace", vmi.Namespace, vmi.Name)
+
+			r, w, _ := os.Pipe()
+			os.Stdout = w
+
+			outC := make(chan string)
+			go func() {
+				var buf bytes.Buffer
+				io.Copy(&buf, r)
+				outC <- buf.String()
+			}()
+
+			// Run this as go routine because to keep proxy open in the background
+			go proxyOnlyCommand.Execute()
+
+			// No simple way to check if proxy has started, so arbitrary delay
+			time.Sleep(10 * time.Second)
+
+			w.Close()
+			out := <-outC
+
+			Expect(out).ToNot(BeNil())
+
+			var result map[string]interface{}
+			err := json.Unmarshal([]byte(out), &result)
```

Comment:
Json can be directly read from the pipe: `err := json.NewDecoder(r).Decode(&result)`.

It may be useful to add a timeout for the read, in case the pipe is not written to because of some error. To do it, the read can be moved to a new goroutine, and then use a select statement to wait on the result or error.

Reply:
Added an `Eventually` block for this